src/graphql/root.rs

```diff
@@ -395,7 +395,14 @@ graphql_object!(Query: Context |&self| {
 
     /// Find all the message types we know about
     field messageTypes(&executor) -> FieldResult<Vec<String>>{
-        Err("Not implemented")?
+        let connection = executor.context().connection.lock()?;
```

Comment:
Not sure about the `lock()`. This is how it's done for all the queries, but from what I understand, sqlite3 with Write Ahead Logging (WAL) doesn't need it.

Reply:
For now we still need to use a Mutex on the db connection. If we don't, things explode at runtime when they can't get db access because it's locked.

I agree that we do want to get the wal working. But it has a few steps:

- we need to be able to pass configuration to sqlite so that we can set the threading mode to `2`
- enable wal mode, which is easy, can be done with: `PRAGMA journal_mode=wal`
- create pools of read-only and write connections.